pygfx/geometries/_text.py

```diff
@@ -0,0 +1,167 @@
+
+        self.indices = np.zeros((len(text),), np.uint32)
+        self.positions = np.zeros((len(text), 3), np.float32)
+
+        for i in range(len(text)):
+            self.positions[i] = i * font_size, 0, 0
+
+
+class TextGeometry(Geometry):
+    def __init__(self, text_parts, max_width=0, line_height=1.2, text_align="left"):
+        super().__init__()
+
+        # Check incoming parts
+        self._text_parts = tuple(text_parts)
+        for part in self._text_parts:
+            if not isinstance(part, TextPart):
+                raise TypeError("TextGeometry only accepts TextPart objects.")
+
+        # Set props
+        self.max_width = max_width
+        self.line_height = line_height
+        self.text_align = text_align
+
+        # Compose the parts in a single geometry
+        indices_arrays = []
+        positions_arrays = []
+        glyph_count = 0
+        for part in self._text_parts:
+            assert part.indices.dtype == np.uint32
+            assert part.positions.dtype == np.float32
+            assert part.positions.shape == (part.indices.size, 3)
+            part.offset = glyph_count
+            glyph_count += part.indices.size
+            indices_arrays.append(part.indices)
+            positions_arrays.append(part.positions)
+
+        # Store
+        self.indices = Buffer(np.concatenate(indices_arrays, 0))
+        self.positions = Buffer(np.concatenate(positions_arrays, 0))
+
+    @property
+    def max_width(self):
+        """The maximum width of the text (in model coords). Text will
+        wrap if beyond this limit. Set to 0 for no wrap. Default 0.
+        """
+        return self._max_width
+
+    @max_width.setter
+    def max_width(self, width):
+        self._max_width = float(width or 0)
+
+    @property
+    def line_height(self):
+        """The height of a line of text, used to set the distance between
+        linces. Represented as a factor of the font size. Default 1.2.
+        """
+        return self._line_height
+
+    @line_height.setter
+    def line_height(self, heigh):
+        self._line_height = float(heigh or 1.2)
+
+    @property
+    def text_align(self):
+        """Set the alignment of the text. Can be left, right, center, or justify.
+        Default "left".
+        """
+        return self._text_align
+
+    @text_align.setter
+    def text_align(self, align):
+        alignments = "left", "right", "center", "justify"
+        if align is None:
+            align = "left"  # todo: or does center make more sense in a viz lib?
+        elif isinstance(align, int):
+            align = {-1: "left", 0: "center", 1: "right"}[align]
+        elif not isinstance(align, str):
+            raise TypeError("Align must be a string.")
+        align = align.lower()
+        if align not in alignments:
+            raise ValueError(f"Align must be one of {alignments}")
+        self._text_align = align
+
+    def _position(self):
```

Comment:
I'm imagining you intend to expose these helper functions the geometry to update positions as needed when certain properties are changed, right? Makes sense!

Reply:
Yes indeed. All values that affect positioning can be applied relatively easily *after* the shaping and the whole make-sure-the-sdf-glyph-is-available stuff is done. So it feels like a good idea to make them available as properties. The Geometry and WorldObject are both free of object-specific props so far, so it looks like we need to break that tradition here, and the geometry felt like the best place to do that.

Reply:
I'm fine with putting them on the geometry class for now, it makes most sense.

(Three.js also provides such buffer affecting functions directly on geometry)